### DISCUSSION TITLE
Upload a file with Apollo

### DISCUSSION THREAD

**Discussion by TomasMoskala** [graphql-nexus/nexus#746] [Q&A] [answered]
Hi, 
how do I write a resolver to upload a file and how to use it with Apollo client-server?

import { GraphQLUpload } from 'graphql-upload'
export const Upload = GraphQLUpload

What next?
Thank you.

**Comment by shreyas44** [comment, marked as answer]
@TomasMoskala You can find a ton of resources online regarding file uploads to GraphQL APIs. You can probably start with the [apollo documentation](https://www.apollographql.com/docs/apollo-server/data/file-uploads/#:~:text=Enabling%20file%20uploads%20in%20Apollo%20Server&text=To%20enable%20file%20uploads%2C%20reference,type%20File%20%7B%20filename%3A%20String!) and the [`graphql-upload` docs](https://github.com/jaydenseric/graphql-upload#readme) which you seem to want to use. You could also use [`apollo-server-express`](https://www.npmjs.com/package/apollo-server-express), pass the request as part of the resolver `context` and you can interpret the request like you would a regular express request and get the file and upload it.

However, that being said I would advise against adding file upload to your GraphQL API in a production application for the main reason being that it adds a huge amount of load to the GraphQL server which needs to be super fast and responsive as most of your application would depend on it.

**Reply by TomasMoskala** [reply]
Hi, 
I have the same mindset, I decided to start a new server just for photos and use a simple php upload script leaving apollo just the DB jobs. I will also be able to use any third party for the storage if needed in future.
